railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -141,6 +145,26 @@ def has_uniq_index?
         @has_uniq_index
       end
 
+      def has_default?
+        @has_default
+      end
+
+      def assigned_default
+        case self.type
+        when :string, :text
+          case
+          when @assigned_default[0] == '"' && @assigned_default[-1] == '"'
+            @assigned_default
+          when @assigned_default[0] == "'" && @assigned_default[-1] == "'"
+            @assigned_default
+          else
+            "'#{@assigned_default}'"
+          end
+        else
```

Comment:
This has a lot of gaps in its code, and duplicates  schema dumping logic already present for quoting values.